server/handler/render.go

```diff
@@ -2,49 +2,62 @@ package handler
 
 import (
 	"encoding/json"
+	"fmt"
 	"net/http"
+
+	"github.com/src-d/code-annotation/server/serializer"
+	"github.com/src-d/code-annotation/server/service"
 )
 
-func respOK(d interface{}) response {
-	return response{
-		statusCode: http.StatusOK,
-		Data:       d,
-	}
-}
+func Get(rp RequestProcessFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		response, err := rp(r)
+		// TODO: There should be no ppl calling Write with nil *serializer.Response
+		if response == nil {
+			response = &serializer.Response{}
+		}
 
-func respErr(statusCode int, msg string) response {
-	return response{
-		statusCode: statusCode,
-		Errors:     []errObj{errObj{Title: msg}},
+		Write(w, r, response, err)
 	}
 }
 
-func respInternalErr() response {
-	return respErr(http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
-}
-
-type errObj struct {
-	Title string `json:"title"`
-}
+// TODO: it should be private
```

Comment:
imo, don't need to be private.

Reply:
I made it private because nobody from the outside of the package should write in the response; should it?

Reply:
I can easily imagine when we would use it.
For example, we would create an endpoint to download diff or file.
```go
func GetFile(r, w) {
  name = getFileName(r)
  f, err := getFile(name)
  if err != nil {
    Write(w, Error{"file with name %s doesn't exist", name})
  }
  http.ServeContent(w, f)
}
```

Reply:
but... that endpoint should be a handler, right?
If so, it can use `handler.write` too, as done by `handler.OauthCallback`
https://github.com/src-d/code-annotation/pull/37/commits/76ed47205d7755e30d31fbbdbc26a7381e4de352#diff-3ad2ebeb7ccf861129aa87b26fa1f1d8R31

Reply:
ahhhh. you moved it from serializer to handler. then all good!